src/main/msp/msp.c

```diff
@@ -4185,6 +4185,13 @@ static mspResult_e mspCommonProcessInCommand(mspDescriptor_t srcDesc, int16_t cm
                     // increase the canvas size to the HD default as that is what the user will expect
                     osdConfigMutable()->canvas_cols = OSD_HD_COLS;
                     osdConfigMutable()->canvas_rows = OSD_HD_ROWS;
+                    // Also force use of MSP displayport
+                    osdConfigMutable()->displayPortDevice = OSD_DISPLAYPORT_DEVICE_MSP;
```

Comment:
After making the HD changes, when we know we have HD in the build, we should handle the situation where HD_OSD was not included in the build, by retaining the existing settings
```
#else
                    // must have an SD build option, keep existing SD video_system, do not change canvas size
                    video_system = vcdProfile()->video_system;
#endif
```
If we don't have HD in the build, we must have an SD-only build; keep our existing SD settings